scenarios/command/c/command_client/main.c

```diff
@@ -0,0 +1,205 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <uuid/uuid.h>
+
+#include "mosquitto.h"
+#include "mqtt_callbacks.h"
+#include "mqtt_protocol.h"
+#include "mqtt_setup.h"
+
+#define PAYLOAD "\n\v\b\261Í\244\006\020\364\254\265d\022\nmobile-app"
+#define RESPONSE_TOPIC "vehicles/vehicle03/command/unlock/response"
+#define PUB_TOPIC "vehicles/vehicle03/command/unlock/request"
+#define COMMAND_CONTENT_TYPE "application/protobuf"
+#define COMMAND_TIMEOUT_SEC 10
+
+#define QOS 1
+#define MQTT_VERSION MQTT_PROTOCOL_V5
+
+#define UUID_LENGTH 16
+
+#define CONTINUE_IF_ERROR(rc)                                   \
+  do                                                            \
+  {                                                             \
+    if (rc != MOSQ_ERR_SUCCESS)                                 \
+    {                                                           \
+      printf("Error publishing: %s\n", mosquitto_strerror(rc)); \
+      mosquitto_property_free_all(&proplist);                   \
+      continue;                                                 \
+    }                                                           \
+  } while (0)
+
+static uuid_t pending_correlation_id;
+static time_t last_command_time;
+
+// Custom callback for when a message is received.
+// prints the message information from the command response and validates that the correlation data
+// matches.
+void handle_message(
+    struct mosquitto* mosq,
+    const struct mosquitto_message* message,
+    const mosquitto_property* props)
+{
+  printf(
+      "on_message: Topic: %s; QOS: %d; protobuf Payload: %s\n",
+      message->topic,
+      message->qos,
+      (char*)message->payload);
+
+  void* correlation_data;
+  uint16_t correlation_data_len;
+  if (mosquitto_property_read_binary(
+          props, MQTT_PROP_CORRELATION_DATA, &correlation_data, &correlation_data_len, false)
+      == NULL)
+  {
+    printf("Error reading correlation data\n");
+    return;
+  }
+
+  char readable_correlation_data[correlation_data_len];
```

Comment:
Should `correlation_data_len` always be 37 (to account for trailing zero). 

See: https://www.man7.org/linux/man-pages/man3/uuid_unparse.3.html

Reply:
fixed!